TESTING.adoc

```diff
@@ -0,0 +1,107 @@
+This file imports the test suite from the Minishift repository and can be used for the import of additional step definitions.
+
+[[prerequisites]]
+== Prerequisites
+You must do the following to run the tests successfully:
+
+- Install link:https://docs.openshift.org/latest/minishift/getting-started/installing.html[Minishift and its prerequisites].
+- Install a recent link:https://golang.org/dl/[Go distribution] (>=1.8).
+- Install the link:https://github.com/golang/dep[dep package] with the following: `go get -u github.com/golang/dep/cmd/dep`
+- `git clone` this repository into *_$GOPATH/src/github.com/minishift/minishift-addons_*.
+- Get the latest `minishift` binary from https://github.com/minishift/minishift/releases/latest and place it in *_testing/bin_*.
+- Vendor the needed Go packages by runining `make vendor` in the root directory of the *minishift-addons* repository.
+
+[NOTE]
+====
+We recommend vendoring the packages after every `git pull` from the *minishift-addons* master branch.
+====
+
+[[running-tests]]
+== Running Tests
+To run integration tests for a specific add-on, go to the root directory of the *minishift-addons* repository and run the following command: 
+
+```
+make integration ADDON=<name-of-addon>
+```
+
+This command will start tests on all Gherkin feature files present in *_tests/integration/features_* which contain the tag `@<name-of-addon>`.
+For most add-ons, this will mean using one feature file located at *_tests/integration/features/<name-of-addon>.feature_*.
+
+To run the tests on all add-ons, you can use the following:
+
+```
+make integration
+```
+
+[[additional-options]]
+=== Additional Options
+It is possible to run the tests with a Minishift binary located outside of the default location of *_testing/bin_*.
+To do so, use the `MINISHIFT_BINARY` parameter with the `make integration` target.
+For example:
+
+```
+make integration ADDON=xpaas MINISHIFT_BINARY=/home/user/minishift/minishift
+```
+
+[[writing-tests]]
+== Writing Tests
+Create a new file named *_tests/integration/features/<name-of-add-on>.feature_* to create a new integration test for an add-on.
+The *_example.feature_* file demonstrates the basic structure of an integration test for an example add-on and may be used as a starting point.
+
+Features should contain at least 6 key parts which secure that the tested add-on can be installed, applied, removed and uninstalled:
+
+- Installation of the add-on: `executing "minishift addons install ../../add-ons/<name-of-addon>" succeeds`
+- Starting of Minishift: `executing "minishift start" succeeds`
+- Application of the add-on: `executing "minishift addons apply <name-of-addon>" succeeds`
+- Removal of the add-on: `executing "minishift addons remove <name-of-addon>" succeeds`
+- Add-on uninstallation: `executing "minishift addons uninstall <name-of-addon>" succeeds`
+- Deletion of the Minishift instance: `executing "minishift delete --force" succeeds`
+
+[[advanced-steps]]
```

Comment:
This is what I was talking about documentation. 
I would be happy if we list all possible steps one can use to encourage the add-on developers to write tests.
I have a hard time in assuming users will go to testsuite package and read the go code(also regex magic) to see all the available steps.

It should have been in a new doc file.

Reply:
@budhrg Agree. I am thinking about it for some time even at Minishift repository, I would like to take it as part of https://github.com/minishift/minishift/issues/1423. However for this PR I think we could be fine without it - still one can take a look on existing feature files and re-use parts from them. I can add more feature files for addons we currently have in repo and try to test more deeply where I would be able to do so. Maybe this could help the contributors, wdyt?

Reply:
> I am thinking about it for some time even at Minishift repository, I would like to take it as part of minishift/minishift#1423

That is different issue altogether I guess. Issue 1423 is to document tags. 
We can create another issue to document all available steps.

> I can add more feature files for addons we currently have in repo.

This is not part of this issue I guess and for some addon it might take some more time like Istio.